src/runtime/storageNG/storage-proxy.ts

```diff
@@ -123,13 +124,19 @@ export class StorageProxy<T extends CRDTTypeRecord> {
   }
 
   async getParticleView(): Promise<[T['consumerType'], VersionMap]> {
-    await this.synchronizeModel();
-    return [this.crdt.getParticleView()!, this.versionCopy()];
-  }
-
-  async getData(): Promise<T['data']> {
-    await this.synchronizeModel();
-    return this.crdt.getData();
+    if (this.synchronized) {
+      return [this.crdt.getParticleView()!, this.versionCopy()];
+    } else {
+      const promise: Promise<[T['consumerType'], VersionMap]> =
+          new Promise((resolve) => {
+            this.modelHasSynced = () => {
+              this.modelHasSynced = () => undefined;
+              resolve([this.crdt.getParticleView()!, this.versionCopy()]);
+            };
+          });
+      await this.synchronizeModel();
```

Comment:
the name synchronizeModel sounds like the function actually completes synchronization; that would mean awaiting it here would remove the need for a promise at all. So we should either comment the usage here to point out that the call only sends the synchronize message; or change the name of synchronizeModel to something like requestSynchronization.